Emails: Fix Add new mailboxes link not shown for email forwards on Email Plan page

stephanethomas opened the pull request:
This pull request fixes a regression introduced in https://github.com/Automattic/wp-calypso/pull/53189, which would prevent the `Add new mailboxes` option from being displayed for email forwards on the `Email Plan` page. This patch fixes that issue, and also:

* Renames `Add new mailboxes` to `Add new email forwards`
* Removes `domain.currentUserCanManage` checks which are specific to domains (and not emails)
* Removes some unnecessary checks on `domain` and `selectedSite`

Before | After
------ | -----
<img width="624" alt="02" src="https://user-images.githubusercontent.com/594356/120495547-18f6e100-c3bd-11eb-9d8f-7158909c527b.png"> |<img width="624" alt="01" src="https://user-images.githubusercontent.com/594356/120495561-1c8a6800-c3bd-11eb-9a81-a38f29322d26.png">

#### Testing instructions

1. Run `git checkout fix/adding-email-forwards` and start your server, or open a [live branch](https://calypso.live/?branch=fix/adding-email-forwards)
2. Log into a WordPress.com account that has a domain with email forwards
3. Open the [`Emails` page](http://calypso.localhost:3000/email)
4. Click the domain with email forwards
5. Assert that a `Add new email forwards` link is displayed
6. Click that link
7. Assert that you are presented with the `Email Forwarding` page

stephanethomas committed:
Emails: Fix Add new mailboxes link not shown for email forwards on Email Plan page

stephanethomas commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -267,16 +266,24 @@ class EmailPlan extends React.Component {
 		}
 
-		if ( ! selectedSite || ! purchase ) {
+		if ( ! purchase ) {
```
It looks like this change I've recommend in https://github.com/Automattic/wp-calypso/pull/53189/files#r643073055 wasn't applied in the end.

olaseni replied:
I remember applying that change. I must have missed pushing that up.

stephanethomas commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -246,10 +249,6 @@ class EmailPlan extends React.Component {
-		if ( ! domain ) {
-			return null;
-		}
-
```
`domain` is always provided to this component so we can remove that condition.

olaseni commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -267,16 +266,24 @@ class EmailPlan extends React.Component {
 			return <VerticalNavItem isPlaceholder />;
 		}
 
+		if ( ! hasSubscription ) {
```
We should probably make a case for MVP accounts, because they do not actually have a subscription?

stephanethomas committed:
Emails: Show Add new mailboxes link on Email Plan page for Titan accounts with no subscription

olaseni commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -267,13 +266,17 @@ class EmailPlan extends React.Component {
 		}
 
-		if ( ! selectedSite || ! purchase ) {
+		if ( ! purchase ) {
```
Did you check with an MVP account? I think no subscription also means no purchase, hence the placeholder may be there forever.

stephanethomas replied:
No I didn't, I don't have such account.

daledupreez commented on client/my-sites/email/email-management/home/email-plan.jsx:
```diff
@@ -267,13 +266,17 @@ class EmailPlan extends React.Component {
 		}
 
-		if ( ! selectedSite || ! purchase ) {
+		if ( ! purchase ) {
```
This logic still doesn't feel right -- we won't have a `purchase` for all MVP subscriptions, so we should not show a purchase for that case. I think the logic should be adjusted slightly:
```
if ( hasSubscription ) { // assuming this can be resolved immediately and is not dependent on async data loading
  if ( ! purchase ) {
    // return placeholder
  }

  if ( isExpired( purchase ) ) {
    // return renewal message
  }
}

// return add new mailboxes content
```

stephanethomas committed:
Emails: Refine logic to display Add new mailboxes link on Email Plan page

stephanethomas commented:
I've updated the `renderAddNewMailboxesOrRenewNavItem()` method per your feedback. Note I've refactored its logic in order to be consistent with other methods from the `EmailPlan` component that treat email forwards as the default option if the domain neither has G Suite nor Titan. I also wanted to separate clearly the case where a Titan account has no subscriptions.

olaseni reviewed (approved):
LGTM! Works for all tripartite scenarios of Google Workspace, Professional Email and Forwards

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5977448

Thank you @stephanethomas for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5977448).
